pkg/app/utils/utils.go

```diff
@@ -125,3 +123,32 @@ func DeployApp(mgmtAppClient projv3.AppInterface, projectID string, createOrUpda
 
 	return rtn, nil
 }
+
+func DetectAppCatalogExistence(appCatalogID string, cattleTemplateVersionsClient v3.CatalogTemplateVersionInterface) error {
+	templateVersionID, templateVersionNamespace, err := common.ParseExternalID(appCatalogID)
+	if err != nil {
+		return errors.Wrapf(err, "failed to parse catalog ID %q", appCatalogID)
+	}
+
+	_, err = cattleTemplateVersionsClient.GetNamespaced(templateVersionNamespace, templateVersionID, metav1.GetOptions{})
```

Comment:
Lister?

Reply:
That's existing code ;-) didn't want to touch it